src/notebook.jl

```diff
@@ -173,6 +173,35 @@ macro ingredients(filename)
 	end
 end
 
+# ╔═╡ 36a67c48-f398-4ec1-9f01-b33d647137ee
+"""
+Run a code block using debounce, i.e. only run it when variable list has stabilized through `timeout`.
+
+```julia
+@use_debounce([var], 1) do
+	sleep(2.)
+	@info "stable variable \$var"
+end
+```
+"""
+macro use_debounce(f, deps, timeout)
+	quote
+		ref, set_ref = @use_state(nothing)
+		count_ref = @use_ref(0)
+
+		count_ref.x += 1
+		task = Task(function()
+			count_copy = deepcopy(count_ref.x)
+			sleep($timeout)
+			if count_copy == count_ref.x
+				set_ref($(esc(f))())
+			end
+		end)
+		schedule(task)
+		ref	
```

Comment:
Since this is meant to be used in similar cases as `@use_memo`, i think it should have the same api and return the value, not the ref itself.
```suggestion
		ref[]
```

Reply:
I think I should actually change the variable name to something in the line of `value`, since this isn't a `ref` but a state value.